core/src/main/java/tachyon/master/TachyonMaster.java

```diff
@@ -186,20 +185,13 @@ boolean isZookeeperMode() {
     return mZookeeperMode;
   }
 
-  private void login() throws IOException {
-    MasterConf mConf = MasterConf.get();
-    if (mConf.KEYTAB == null || mConf.PRINCIPAL == null) {
-      return;
-    }
+  private void connect() throws IOException {
     UnderFileSystem ufs = UnderFileSystem.get(CommonConf.get().UNDERFS_ADDRESS);
-    if (ufs instanceof UnderFileSystemHdfs) {
-      ((UnderFileSystemHdfs) ufs).login(mConf.KEYTAB_KEY, mConf.KEYTAB, mConf.PRINCIPAL_KEY,
-          mConf.PRINCIPAL, NetworkUtils.getFqdnHost(mMasterAddress));
-    }
+    ufs.connectFromMaster(NetworkUtils.getFqdnHost(mMasterAddress));
```

Comment:
Since this is HDFS only, this logic should only be in the HDFS class.  Can you move that there?  I thought that that was the case before, but guess that didn't happen...